### PR TITLE
Rename the 'Practitioner' Authorization Type to 'Provider'

### PR DESCRIPTION
**Pull request by FFSol**
Description
----

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

Screen Shot 2020-07-23 at 10 53 20 AM

### CODE REVIEW

**Comment by ebaranowski** [comment]
Agree

File: spec/lib/resources_spec.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# frozen_string_literal: true
+provider# frozen_string_literal: true
```
Comment (joeratt) [inline spec/lib/resources_spec.rb:1]: Copy-paste issue? I think this is breaking the tests

File: lib/resources.rb [hunk 2]
```diff
@@ -108,13 +108,13 @@ def get_resource(key)
         end
       end
 
-      def authorization_types(practitioner: true, patient: false, system: false)
+      def authorization_types(provider: true, patient: false, system: false)
```
Comment (MaxPhilips) [inline lib/resources.rb:111]: let's default provider to false if we're passing it in anyways on each method call

either that or actually take advantage of the default

File: lib/resources.rb [hunk 3]
```diff
@@ -108,13 +108,13 @@ def get_resource(key)
...
         html = '<div class="auth-types">'
         prev = false
 
-        if practitioner
-          html += '<a href="/authorization/#requesting-authorization-on-behalf-of-a-user" class="practitioner">'\
-                  'Practitioner</a>'
+        if provider
```
Comment (MaxPhilips) [inline lib/resources.rb:115]: while we're here, what do we think of beautifying this to the following:

```
      def authorization_types(provider: true, patient: false, system: false)
        template = '<a href="/authorization/#requesting-authorization-on-behalf-of-a-%s" class="%s">%s</a>'

        auth_types = []

        auth_types << template % %w[user provider Provider] if provider
        auth_types << template % %w[user patient Patient] if patient
        auth_types << template % %w[system system System] if system

        "<div class=\"auth-types\">#{auth_types.join(' | ')}</div>"
      end
```

File: spec/lib/resources_spec.rb (hunk 1 above)
Comment (MaxPhilips) [reply spec/lib/resources_spec.rb]: https://github.com/cerner/fhir.cerner.com/pull/430/commits/56870f0ff4573ad640d54226948788ec51129ab0#diff-0c4be351aacc6e5786906aa41d053668R1

File: lib/resources.rb (hunk 2 above)
Comment (MaxPhilips) [reply lib/resources.rb]: defaulting to false https://github.com/cerner/fhir.cerner.com/pull/430/commits/56870f0ff4573ad640d54226948788ec51129ab0#diff-7718c11ed130ac00a9854e0b089f74b4R111

File: lib/resources.rb (hunk 3 above)
Comment (MaxPhilips) [reply lib/resources.rb]: https://github.com/cerner/fhir.cerner.com/pull/430/commits/56870f0ff4573ad640d54226948788ec51129ab0#diff-7718c11ed130ac00a9854e0b089f74b4R112

File: lib/resources.rb (hunk 2 above)
Comment (MaxPhilips) [reply lib/resources.rb]: need to fix up some specs

Comment (MaxPhilips) [reply lib/resources.rb]: let's add an .rspec file as well

Comment (MaxPhilips) [reply lib/resources.rb]: https://github.com/cerner/fhir.cerner.com/pull/430/commits/204ced2476ff61648bb6e80f9cebee4c95f0aeab

**Comment by MaxPhilips** [comment]
after rebasing, we need to switch the param name in newly added observation files

**Comment by MaxPhilips** [comment]
> after rebasing, we need to switch the param name in newly added observation files

done f119e8d

File: lib/resources.rb (hunk 3 above)
Comment (MaxPhilips) [reply lib/resources.rb]: fixed lint a040879

**Comment by FFSol** [comment]
Re-reviewed, re-validated.

**Comment by FFSol** [comment]
Confirmed live: 
Screen Shot 2020-08-27 at 4 06 26 PM

**Pull request merged**